### ISSUE TITLE
Coverage not working on travis-ci

### ISSUE THREAD

**Issue by holgerteichgraeber** [JuliaCI/Coverage.jl#228]
Thanks for a great package! 
I am trying to integrate Coverage and codecov.io into my package (`ClustForOpt.jl`).

Even though I have added the required line into my `travis.yml` file, and my test folder is set up similarly to the ones for example in `GeoStats` or `JuMP`, nothing with relation to testing is working. 

See here on my testing branch: [https://github.com/holgerteichgraeber/ClustForOpt.jl/blob/testing/.travis.yml](https://github.com/holgerteichgraeber/ClustForOpt.jl/blob/testing/.travis.yml) , and the resulting run on travis-ci: [https://travis-ci.com/holgerteichgraeber/ClustForOpt.jl/builds/117243942](https://travis-ci.com/holgerteichgraeber/ClustForOpt.jl/builds/117243942).

What am I missing to get Coverage.jl running?

**Comment by fingolfin** [comment]
Looking at the end of the log of the job you pasted, it says: "Skipping the after_success step, as specified in the configuration.". And indeed, your config has `after_success: skip`. But `after_success` processes the coverage...

**Comment by holgerteichgraeber** [comment]
Sounds good, I just changed it to `after_success:      ` and left out the `skip`, let's see what happens..

**Comment by holgerteichgraeber** [comment]
It seems that it didn't make a difference in the run: [https://travis-ci.com/holgerteichgraeber/ClustForOpt.jl/jobs/211855676](https://travis-ci.com/holgerteichgraeber/ClustForOpt.jl/jobs/211855676). 

Any other thoughts?

Somehow Coverage/test, none of the like show up in the travis-ci run..

**Comment by fingolfin** [comment]
Now you provided a custom, empty `after_success:` for your job. You have to remove that erroneous `after_success:` in line 20.

**Comment by holgerteichgraeber** [comment]
Thanks @fingolfin . Coverage now appears in travis-ci. It didn't run any tests though. Do I need to add anything, run tests seperately? I thought Coverage.jl was going to run the tests in `runtests.jl` automatically and then report that to codecov?

**Comment by holgerteichgraeber** [comment]
This is the travis run:  https://travis-ci.com/holgerteichgraeber/ClustForOpt.jl/jobs/211869167

**Comment by fingolfin** [comment]
No it does not run the tests for you. I would recommend that you first read the README carefully, then look at some projects and their `.travis.yml` which use `Coverage.jl`, such as `Coverage.jl`, and some of the dozens of other projects listed in the README.

**Comment by holgerteichgraeber** [comment]
Thanks @fingolfin . Yes, I looked at several of the projects such as `Geostats.jl` and `JuMP.jl`, which both use `Coverage.jl`.  
I understand the travis-ci documentation that it enables coverage automatically: https://docs.travis-ci.com/user/languages/julia/#default-build-and-test-script 
It says the default build is 
> using Pkg
> Pkg.build() # Pkg.build(; verbose = true) for Julia 1.1 and up
> Pkg.test(coverage=true)
So there is no need to enable testing with coverage separately anywhere?

Based on the documentation of `Coverage.jl`, to me my `travis.yml` script, where I just added the lines outlined in `Coverage.jl` documentation, should be enough to get coverage reports. The lines are:
> after_success:
> - julia -e 'using Pkg; Pkg.add("Coverage"); using Coverage; Codecov.submit(process_folder())'

Why am I not getting any coverage reports?
